feature/appdetail/src/test/kotlin/com/merxury/blocker/feature/appdetail/AppDetailViewModelTest.kt

```diff
@@ -240,10 +242,240 @@ class AppDetailViewModelTest {
         collectJob1.cancel()
+        collectJob2.cancel()
+    }
+
+    @Test
+    fun appBarUiState_whenSearchResult_thenShowSearchResult() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.appInfoUiState.collect() }
+        appRepository.sendAppList(sampleAppList)
+        userDataRepository.sendUserData(defaultUserData)
+        viewModel.loadAppInfo()
+        componentRepository.sendComponentList(sampleComponentList)
+        componentDetailRepository.sendComponentDetail(sampleComponentDetailList)
+        viewModel.loadComponentList()
+        viewModel.changeSearchMode(true)
+        val keyword = "Activity"
+        viewModel.search(keyword)
+        assertEquals(
+            Result.Success(
+                ComponentSearchResult(
+                    app = sampleAppList.first().toAppItem(),
+                    activity = sampleComponentList.filter {
+                        it.type == ACTIVITY && it.name.contains(
+                            keyword,
+                            ignoreCase = true,
+                        )
+                    },
+                    service = sampleComponentList.filter {
+                        it.type == SERVICE && it.name.contains(
+                            keyword,
+                            ignoreCase = true,
+                        )
+                    },
+                    receiver = sampleComponentList.filter {
+                        it.type == RECEIVER && it.name.contains(
+                            keyword,
+                            ignoreCase = true,
+                        )
+                    },
+                    provider = sampleComponentList.filter {
+                        it.type == PROVIDER && it.name.contains(
+                            keyword,
+                            ignoreCase = true,
+                        )
+                    },
+                ),
+            ),
+            viewModel.appInfoUiState.value.componentSearchUiState,
+        )
+        collectJob.cancel()
+    }
+
+    @Test
+    fun appBarUiState_whenExitSearchMode_thenShowAllComponents() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.appInfoUiState.collect() }
+        appRepository.sendAppList(sampleAppList)
+        userDataRepository.sendUserData(defaultUserData)
+        viewModel.loadAppInfo()
+        componentRepository.sendComponentList(sampleComponentList)
+        componentDetailRepository.sendComponentDetail(sampleComponentDetailList)
+        viewModel.loadComponentList()
+        viewModel.search("Activity")
+        viewModel.changeSearchMode(true)
+        viewModel.changeSearchMode(false)
+        assertEquals(
+            Result.Success(
+                ComponentSearchResult(
+                    app = sampleAppList.first().toAppItem(),
+                    activity = sampleComponentList.filter { it.type == ACTIVITY },
```

Comment:
Filter by name, not by type

Reply:
It's because I need to fill the data list into ComponentSearchResult